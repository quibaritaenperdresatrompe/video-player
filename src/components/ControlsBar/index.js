import glamorous from 'glamorous'
import PropTypes from 'prop-types'
import React, { Component } from 'react'

import Icon from '../../components/Icon'

const ControlsBarContainer = glamorous.div({
  display: 'flex',
  alignItems: 'center',
})

const ControlContainer = glamorous.span({
  cursor: 'pointer',
  margin: '0 0.5em',
})

const TimeContainer = glamorous.span({
  fontSize: '0.6em',
  margin: '0 0.5em',
})

const secondsToTime = seconds => new Date(seconds * 1e3).toISOString().substr(11, 8)

class ControlsBar extends Component {
  getActionIcon = () => {
    const {
      isPlaying,
      isComplete,
    } = this.props

    if (isComplete) return 'ReplayIcon'
    if (isPlaying) return 'PauseIcon'
    return 'PlayIcon'
  }

  renderAction = () => {
    const {
      isPlaying,
      isComplete,
      pause,
      play,
    } = this.props

    const ActionIcon = this.getActionIcon()

    return (
      <ControlContainer onClick={isPlaying && !isComplete ? pause : play}>
        <Icon
          iconName={ActionIcon}
          size='1.2em'
        />
      </ControlContainer>
    )
  }

  renderTime = () => {
    const currentTime = secondsToTime(this.props.currentTime)
    const totalTime = secondsToTime(this.props.duration)
    return (
      <TimeContainer>
        {`${currentTime} / ${totalTime}`}
      </TimeContainer>
    )
  }

  render() {
    const {
      pause,
      play,
    } = this.props

    if (!pause && !play) return null
    return (
      <ControlsBarContainer>
        {this.renderAction()}
        {this.renderTime()}
      </ControlsBarContainer>
    )
  }
}

ControlsBar.propTypes = {
  currentTime: PropTypes.number,
  duration: PropTypes.number,
  isComplete: PropTypes.bool,
  isPlaying: PropTypes.bool,
  pause: PropTypes.func,
  play: PropTypes.func,
}

ControlsBar.defaultProps = {
  currentTime: 0,
  duration: 0,
  isComplete: false,
  isPlaying: false,
  pause: null,
  play: null,
}

export default ControlsBar
