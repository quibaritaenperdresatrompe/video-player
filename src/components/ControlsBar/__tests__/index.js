import ControlsBar from '../'

import itRendersAllMutations from '../../../utils/jest-it-renders-all-mutations'

const mutations = [
  {
    name: 'without props',
    props: null,
  },
  {
    name: 'with all props',
    props: {
      currentTime: 99,
      duration: 999,
      pause: jest.fn(),
      play: jest.fn(),
    },
  },
  {
    name: 'without pause prop',
    props: {
      currentTime: 99,
      duration: 999,
      play: jest.fn(),
    },
  },
  {
    name: 'without play prop',
    props: {
      currentTime: 99,
      duration: 999,
      pause: jest.fn(),
    },
  },
]

describe(ControlsBar.name, () => {
  itRendersAllMutations(ControlsBar, mutations)
})
