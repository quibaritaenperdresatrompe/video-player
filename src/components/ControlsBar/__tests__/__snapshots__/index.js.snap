// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsBar when it renders pause button it renders correctly 1`] = `
<div
  className="css-12alag6"
>
  <span
    className="css-bcxn3o"
    onClick={[MockFunction]}
  >
    <svg
      className="css-gt3xp3"
      color="hsla(0, 0%, 100%, 0.8)"
      height="1.2em"
      viewBox="0 0 24 24"
      width="1.2em"
    >
      <path
        d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    className="css-1g6ztab"
  >
    00:00:00 / 00:00:00
  </span>
</div>
`;

exports[`ControlsBar when it renders play button it renders correctly 1`] = `
<div
  className="css-12alag6"
>
  <span
    className="css-bcxn3o"
    onClick={[MockFunction]}
  >
    <svg
      className="css-gt3xp3"
      color="hsla(0, 0%, 100%, 0.8)"
      height="1.2em"
      viewBox="0 0 24 24"
      width="1.2em"
    >
      <path
        d="M8 5v14l11-7z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    className="css-1g6ztab"
  >
    00:00:00 / 00:00:00
  </span>
</div>
`;

exports[`ControlsBar when it renders replay button it renders correctly 1`] = `
<div
  className="css-12alag6"
>
  <span
    className="css-bcxn3o"
    onClick={[MockFunction]}
  >
    <svg
      className="css-gt3xp3"
      color="hsla(0, 0%, 100%, 0.8)"
      height="1.2em"
      viewBox="0 0 24 24"
      width="1.2em"
    >
      <path
        d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    className="css-1g6ztab"
  >
    00:00:00 / 00:00:00
  </span>
</div>
`;

exports[`ControlsBar when it renders time it renders correctly 1`] = `
<div
  className="css-12alag6"
>
  <span
    className="css-bcxn3o"
    onClick={[MockFunction]}
  >
    <svg
      className="css-gt3xp3"
      color="hsla(0, 0%, 100%, 0.8)"
      height="1.2em"
      viewBox="0 0 24 24"
      width="1.2em"
    >
      <path
        d="M8 5v14l11-7z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    className="css-1g6ztab"
  >
    00:01:39 / 00:16:39
  </span>
</div>
`;

exports[`ControlsBar without props it renders correctly 1`] = `null`;
